### PR TITLE
Update Terraform Install Docs

### PR DESCRIPTION
**Pull request by rssnyder**
# Harness Developer Pull Request

change misleading terraform install documentation.

- we should be recommending customers create a custom delegate image when installing 3d party tools
- the current script targets macos on intel which is a rare case when running a delegate
  - provide tips for changing the script based on the users target architecture

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Comment by mansong1** [comment]
LGTM

**Comment by suranc** [comment]
Change looks great from an SA perspective.  

Though eventually I think it might be worth a more generic page where we can bake in some of the best practices stuff from this change.  Then link back to it from pages like this.  Such as:

* Recommending enterprise customers create their own delegate image, rather than using `INIT_SCRIPT`
* Switching to `root` and then creating a real user for UID 1001, then switching back to 1001 with the code the change adds here:
```
USER 0
RUN useradd -u 1001 -g 0 harness
...
USER 1001
```

**Comment by rssnyder** [comment]
> * Switching to `root` and then creating a real user for UID 1001, then switching back to 1001 with the code the change adds here:

This will be fixed in the next delegate image, then I will remove that part.

**Comment by suranc** [comment]
Great news!

File: docs/continuous-delivery/cd-advanced/terraform-category/terraform-how-tos.md [hunk 1]
```diff
@@ -26,15 +26,52 @@ Terraform must be installed on the Delegate to use a Harness Terraform Provision
 
 See [Build custom delegate images with third-party tools](/docs/platform/2_Delegates/customize-delegates/build-custom-delegate-images-with-third-party-tools.md).
 
+### Target OS and Architecture
```
Comment (michaelcretzman) [inline docs/continuous-delivery/cd-advanced/terraform-category/terraform-how-tos.md:29]:
```suggestion
### Target OS and architecture
```

File: docs/continuous-delivery/cd-advanced/terraform-category/terraform-how-tos.md [hunk 2]
```diff
@@ -26,15 +26,52 @@ Terraform must be installed on the Delegate to use a Harness Terraform Provision
...
+
+You will need to change the following scripts based on the operating system and architecture of the machine that is running the delegate.
+
+Linux on x86(Intel or AMD): linux_amd64
+Linux on Arm: linux_arm64
+MacOs on x86(Intel): darwin_amd64
+MacOs on ARM(Apple): darwin_arm64
```
Comment (michaelcretzman) [inline docs/continuous-delivery/cd-advanced/terraform-category/terraform-how-tos.md:36]:
```suggestion
- Linux on x86(Intel or AMD): linux_amd64
- Linux on Arm: linux_arm64
- MacOs on x86(Intel): darwin_amd64
- MacOs on ARM(Apple): darwin_arm64
```

File: docs/continuous-delivery/cd-advanced/terraform-category/terraform-how-tos.md (hunk 1 above)
Comment (michaelcretzman) [reply docs/continuous-delivery/cd-advanced/terraform-category/terraform-how-tos.md:29]: we use sentence case for headings: https://harness.atlassian.net/wiki/spaces/DOCS/pages/21130772574/Documentation+style+guide

File: docs/continuous-delivery/cd-advanced/terraform-category/terraform-how-tos.md [hunk 3]
```diff
@@ -26,15 +26,52 @@ Terraform must be installed on the Delegate to use a Harness Terraform Provision
...
+  && unzip -q terraform_1.3.1_linux_amd64.zip \
+  && mv ./terraform /usr/bin/ \
+  && terraform --version
+
+USER 1001
+```
+
+### Init Script
+
+Alternativly you can install terraform when a delegate pod is launched by specifying an initilization script in the delegates environment.
```
Comment (michaelcretzman) [inline docs/continuous-delivery/cd-advanced/terraform-category/terraform-how-tos.md:62]:
```suggestion
### Init script

Alternatively, you can install Terraform when a delegate pod is launched by specifying an initialization script in the delegate YAML `INIT_SCRIPT` [environment variable](https://developer.harness.io/docs/platform/delegates/delegate-reference/delegate-environment-variables/).
```

**Review by michaelcretzman — CHANGES_REQUESTED** [review]
@rssnyder please incorp my suggestions and then I can approve and merge.

Please watch for correct spelling, markdown, and our house style in the future. Thanks

style guide: https://harness.atlassian.net/wiki/spaces/DOCS/pages/21130772574/Documentation+style+guide

**Pull request merged**